syntax="proto3";

// service data_transfer{
//   rpc ClientSendStream(stream StreamField)returns(ClientSendReply){}
//   rpc ClientSend(StreamField)returns(ClientSendReply){}
// }
//流数据自定义字段
message StreamField{
  sint64 timeStamp=1; //时间戳
  string from_=2;
  string to=3;
  string spaceName=4;
  bytes specialField=5;//其他用户自定义字段，通过proto文件序列化后的字节流
}

message Node{
    string address=1;//账户地址                            StructField('address',StringType(),True),\n",
    sint64 total_transactions=2;//总交易量                            StructField('Total_transactions',IntegerType(),True),\n",
    int32 number_of_received_address=3;//                           StructField('Number_of_received_addresses',IntegerType(),True),\n",
    int32 number_of_sent_address=4;//                       StructField('Number_of_sent_addresses',IntegerType(),True),\n",
    int32 created_contracts=5;//                            StructField('Creatd_Contracts',IntegerType(),True),\n",
    double min_value_received=6;//                           StructField('Min_value_received',doubleType(),True),\n",
    double max_value_received=7;//                            StructField('Max_value_received',doubleType(),True),\n",
    double avg_value_received=8;//                            StructField('Avg_value_received',doubleType(),True),\n",
    double total_ether_sent_for_accounts=9;//                            StructField('Total_ether_sent_for_accounts',doubleType(),True),\n",
    double min_value_sent=10;//                            StructField('Min_value_sent',doubleType(),True),\n",
    double max_value_sent=11;//                           StructField('Max_value_sent',doubleType(),True),\n",
    double avg_value_sent=12;//                           StructField('Avg_value_sent',doubleType(),True),\n",
    double total_ether_received_for_accounts=13;//"                            StructField('Total_ether_received_for_accounts',doubleType(),True),\n",
    double min_value_sent_to_contracts=14;//                           StructField('Min_value_sent_to_contracts',doubleType(),True),\n",
    double max_value_sent_to_contracts=15;//                           StructField('Max_value_sent_to_contracts',doubleType(),True),\n",
    double avg_value_sent_to_contracts=16;//                            StructField('Avg_value_sent_to_contracts',doubleType(),True),\n",
    double total_ether_sent_for_contracts=17;//                            StructField('Total_ether_sent_to_contracts',doubleType(),True),\n",
    double USDT_sent_to_contract=18;//                           StructField('USDT(sent_to_contract)',doubleType(),True),\n",
    double BNB_sent_to_contract=19;//                           StructField('BNB(sent_to_contract)',doubleType(),True),\n",
    double UNI_sent_to_contract=20;//                           StructField('UNI(sent_to_contract)',doubleType(),True),\n",
    double USDC_sent_to_contract=21;//                           StructField('USDC(sent_to_contract)',doubleType(),True),\n",
    double LINK_sent_to_contract=22;//                          StructField('LINK(sent_to_contract)',doubleType(),True),\n",
    double MATIC_sent_to_contract=23;//                            StructField('MATIC(sent_to_contract)',doubleType(),True),\n",
    double HEX_sent_to_contract=24;//                            StructField('HEX(sent_to_contract)',doubleType(),True),\n",
    double BUSD_sent_to_contract=25;//                           StructField('BUSD(sent_to_contract)',doubleType(),True),\n",
    double WBTC_sent_to_contract=26;//                            StructField('WBTC(sent_to_contract)',doubleType(),True),\n",
    double ACA_sent_to_contract=27;//                            StructField('ACA(sent_to_contract)',doubleType(),True),\n",
    double AAVE_sent_to_contract=28;//                            StructField('AAVE(sent_to_contract)',doubleType(),True),\n",
    double SAI_sent_to_contract=29;//                           StructField('SAI(sent_to_contract)',doubleType(),True),\n",
    double DAI_sent_to_contract=30;//                            StructField('DAI(sent_to_contract)',doubleType(),True),\n",
    double SHIB_sent_to_contract=31;//                           StructField('SHIB(sent_to_contract)',doubleType(),True),\n",
    double HT_sent_to_contract=32;//                           StructField('HT(sent_to_contract)',doubleType(),True),\n",
    double MKR_sent_to_contract=33;//                           StructField('MKR(sent_to_contract)',doubleType(),True),\n",
    double CRO_sent_to_contract=34;//                           StructField('CRO(sent_to_contract)',doubleType(),True),\n",
    double COMP_sent_to_contract=35;//                           StructField('COMP(sent_to_contract)',doubleType(),True),\n",
    double LEO_sent_to_contract=36;//                           StructField('LEO(sent_to_contract)',doubleType(),True),\n",
    double FEI_sent_to_contract=37;//                           StructField('FEI(sent_to_contract)',doubleType(),True),\n",
    double USDT_sent=38;//                           StructField('USDT(sent)',doubleType(),True),\n",
    double BNB_sent=39;//                           StructField('BNB(sent)',doubleType(),True),\n",
    double UNI_sent=40;//                           StructField('UNI(sent)',doubleType(),True),\n",
    double USDC_sent=41;//                           StructField('USDC(sent)',doubleType(),True),\n",
    double LINK_sent=42;//                          StructField('LINK(sent)',doubleType(),True),\n",
    double MATIC_sent=43;//                            StructField('MATIC(sent)',doubleType(),True),\n",
    double HEX_sent=44;//                            StructField('HEX(sent)',doubleType(),True),\n",
    double BUSD_sent=45;//                           StructField('BUSD(sent)',doubleType(),True),\n",
    double WBTC_sent=46;//                            StructField('WBTC(sent)',doubleType(),True),\n",
    double ACA_sent=47;//                            StructField('ACA(sent)',doubleType(),True),\n",
    double AAVE_sent=48;//                            StructField('AAVE(sent)',doubleType(),True),\n",
    double SAI_sent=49;//                           StructField('SAI(sent)',doubleType(),True),\n",
    double DAI_sent=50;//                            StructField('DAI(sent)',doubleType(),True),\n",
    double SHIB_sent=51;//                           StructField('SHIB(sent)',doubleType(),True),\n",
    double HT_sent=52;//                           StructField('HT(sent)',doubleType(),True),\n",
    double MKR_sent=53;//                           StructField('MKR(sent)',doubleType(),True),\n",
    double CRO_sent=54;//                           StructField('CRO(sent)',doubleType(),True),\n",
    double COMP_sent=55;//                           StructField('COMP(sent)',doubleType(),True),\n",
    double LEO_sent=56;//                           StructField('LEO(sent)',doubleType(),True),\n",
    double FEI_sent=57;//                           StructField('FEI(sent)',doubleType(),True),\n",
    double USDT_received=58;//                           StructField('USDT(received)',doubleType(),True),\n",
    double BNB_received=59;//                           StructField('BNB(received)',doubleType(),True),\n",
    double UNI_received=60;//                           StructField('UNI(received)',doubleType(),True),\n",
    double USDC_received=61;//                           StructField('USDC(received)',doubleType(),True),\n",
    double LINK_received=62;//                          StructField('LINK(received)',doubleType(),True),\n",
    double MATIC_received=63;//                            StructField('MATIC(received)',doubleType(),True),\n",
    double HEX_received=64;//                            StructField('HEX(received)',doubleType(),True),\n",
    double BUSD_received=65;//                           StructField('BUSD(received)',doubleType(),True),\n",
    double WBTC_received=66;//                            StructField('WBTC(received)',doubleType(),True),\n",
    double ACA_received=67;//                            StructField('ACA(received)',doubleType(),True),\n",
    double AAVE_received=68;//                            StructField('AAVE(received)',doubleType(),True),\n",
    double SAI_received=69;//                           StructField('SAI(received)',doubleType(),True),\n",
    double DAI_received=70;//                            StructField('DAI(received)',doubleType(),True),\n",
    double SHIB_received=71;//                           StructField('SHIB(received)',doubleType(),True),\n",
    double HT_received=72;//                           StructField('HT(received)',doubleType(),True),\n",
    double MKR_received=73;//                           StructField('MKR(received)',doubleType(),True),\n",
    double CRO_received=74;//                           StructField('CRO(received)',doubleType(),True),\n",
    double COMP_received=75;//                           StructField('COMP(received)',doubleType(),True),\n",
    double LEO_received=76;//                           StructField('LEO(received)',FloatType(),True),\n",
    double FEI_received=77;//                           StructField('FEI(received)',FloatType(),True),\n",
    sint64 avg_time_between_sent=78;//                            StructField('Avg_time_between_sent)',IntegerType(),True),\n",
    sint64 avg_time_between_received=79;//                          StructField('Avg_time_between_received',IntegerType(),True),\n",
    sint64 time_between_first_last=80;//                           StructField('Time_between_first_last',IntegerType(),True),\n",
    sint64 ERC20_avg_time_sent=81;//                           StructField('ERC20_avg_time_sent',IntegerType(),True),\n",
    sint64 ERC20_avg_time_received=82;//                            StructField('ERC20_avg_time_received',IntegerType(),True),\n",
    sint64 ERC20_avg_time_contracts=83;//                            StructField('ERC20_avg_time_contracts',IntegerType(),True),\n",
    sint64 indegree=84;//                            StructField('inDegree',IntegerType(),True),\n",
    sint64 outdegree=85;//                            StructField('outDegree',IntegerType(),True),\n",                           StructField('ETH(balance)',FloatType(),True)\n",
}

message Edge{
    sint64 timestamp=1;//                            StructField('timestamp',TimestampType(),True),
    string _from=2;//                        StructField('from',StringType(),True),
    string to=3;//                        StructField('to',StringType(),True),
    string coin=4;//                        StructField('coin',StringType(),True),
    double value=5;//                        StructField('value',FloatType(),True),
    string transHash=6;//                        StructField('transHash',StringType(),True),
    double gasUsed=7;//                        StructField('gasUsed',FloatType(),True),
    double gasLimit=8;//                        StructField('gaslimit',FloatType(),True),
    double fee=9;//                        StructField('fee',FloatType(),True),
    string fromType=10;//                        StructField('fromType',StringType(),True),
    string toType=11;//                        StructField('toType',StringType(),True),
    string transType=12;//                        StructField('transType',StringType(),True),
    int32 isLoop=13;//                        StructField('isLoop',IntegerType(),True),
    int32 status=14;//                        StructField('status',IntegerType(),True)
    int64 id=15;//自增列
    int64 rank=16;//10^9*id+timestamp
    double rate=17;//汇率
    double usd_price=18;//美元价格
}


